### PR TITLE
Configurable sliding window

### PR DESCRIPTION
**Pull request by whoawhoawhoa**
Resolve [issue](https://github.com/fstab/grok_exporter/issues/24) with configurable sliding window for summaries.

### CODE REVIEW

**Comment by whoawhoawhoa** [comment]
Tests went OK on my local machine. I assume it is kind of floating bug.

File: config/v3/configV3.go [hunk 1]
```diff
@@ -211,6 +213,10 @@ func applyImportDefaults(metricConfig *MetricConfig, defaults DefaultConfig) {
 	if metricConfig.Type == "summary" && len(metricConfig.Quantiles) == 0 {
 		metricConfig.Quantiles = defaults.Quantiles
 	}
+	if metricConfig.Type == "summary" && metricConfig.MaxAge == 0 {
+		fmt.Print("set max age default")
```
Comment (fstab) [inline config/v3/configV3.go:217]: please remove :)

**Comment by fstab** [comment]
Looking good, thanks a lot. There's a `fmt.Print()` message left over, please remove it before I merge.

BTW the Windows test on AppVeyor failed because in one of the file tailer tests a goroutine did not shut down properly. This has been flaky for a long time and has nothing to do with your PR. I'd love to know how to fix this though.

File: config/v3/configV3.go (hunk 1 above)
Comment (whoawhoawhoa) [reply config/v3/configV3.go]: Oh, sorry, didn't notice that one.
Removed now.

**Comment by fstab** [comment]
Thanks a lot!

**Pull request merged**